doc/rst/language/spec/statements.rst

```diff
@@ -699,6 +700,65 @@ A ``break`` statement cannot be used to exit a parallel loop
         }
       }
 
+.. _The_Require_statement:
+
+The Require Statement
+---------------------
+
+The require statement provides a means to specify required files from
+within the program. It has an effect similar to adding the specified
+files to the Chapel compiler's command line. The filenames are relative
+to the directory from which the Chapel compiler was invoked. Any
+directories specified using the Chapel compiler's -I or -L flags will
+also be searched for matching files.
+
+.. code-block:: syntax
+
+   require-statement:
+     'require' string-or-identifier-list
+
+   string-or-identifier-list:
+     string-or-identifier
+     string-or-identifier ',' string-or-identifier-list
+
+   string-or-identifier:
+     string-literal
+     identifier
+
+The require statement takes a comma-separated list of param string
+filenames as operands. The filenames must be Chapel source files (*.chpl),
+C source files (*.c), C header files (*.h), precompiled C object
+files (*.o), or precompiled library archives (lib*.a). When using
+precompiled library archives, remove the lib and .a parts of the
+filename and add -l to the beginning as if it were being specified on
+the command line.
+
+.. code-block:: chapel
+
+   require "foo.h", "-lfoo";
+
+Require statements accept general ``param`` string expressions
+beyond the string literals shown in these examples.  Only
+``require`` statements in code that the compiler considers
+executable will be processed.  Thus, a ``require`` statement
+guarded by a ``param`` conditional that the compiler folds out, or
+in a module that does not appear in the program's ``use``
+statements will not be added to the program's requirements.  For
```

Comment:
I observe that the above does not apply to `.chpl` files. That is, a .chpl file is required to be a string literal and is not affected by param folding.

I suggest to consider this an implementation constraint and so does not need to be reflected in the spec. Otherwise it would need to be listed above.